cdap-ui/app/features/admin/templates/namespace/app-metadata.html

```diff
@@ -3,16 +3,30 @@
       <div class="col-xs-6">
           <h2 ng-bind="apps.name | caskCapitalizeFilter"></h2>
       </div>
-      <div class="col-xs-6">
-
-          <div class="text-right">
-            <a class="btn btn-default"
-              data-template="/assets/features/admin/templates/partials/app-dd.html"
-              bs-dropdown>
-                Actions
-                <span class="caret"></span>
-            </a>
-          </div>
+      <div class="col-xs-6 text-right">
+        <div class="btn-group" dropdown="">
+          <a class="btn btn-default dropdown-toggle" dropdown-toggle="" aria-haspopup="true" aria-expanded="false">
+              Actions
+              <span class="caret"></span>
+          </a>
+          <ul class="dropdown-menu dropdown-menu-right" role="menu">
+            <li role="presentation">
+              <a role="menuitem" ui-sref="admin.namespace.detail.apps.metadata.preference">
+                <span class="fa fa-fw fa-cog pull-right"></span>
+                Preferences
+              </a>
+            </li>
+            <li role="presentation">
+              <a role="menuitem" href=""
+                ng-click="caskConfirm()"
+                cask-confirmable="deleteApp(app.id)"
```

Comment:
In the context of "All Applications" app.id will be null. This is delete behavior/button should be added either to all the rows of the app or should atleast link to the admin page that allows the user to delete the app (which is the desired behavior I believe)